lib/Translation/ExportVerilog/ExportVerilog.cpp

```diff
@@ -636,17 +668,29 @@ void EmitterBase::emitTextWithSubstitutions(
       }
       next += 2;
 
-      if (operandNo >= op->getNumOperands()) {
-        emitError(op, "operand " + llvm::utostr(operandNo) + " isn't valid");
-        continue;
-      }
+      Value emitOp;
 
       // Emit any text before the substitution.
       os << string.take_front(start - 2);
-
-      // Emit the operand.
-      operandEmitter(op->getOperand(operandNo));
-
+      // operantNo can either refer to Operands or symOps. Assumption is symOps
+      // are sequentially referenced after the operands.
+      if (operandNo < op->getNumOperands())
+        // Emit the operand.
+        operandEmitter(op->getOperand(operandNo));
+      else if ((operandNo - op->getNumOperands()) < numSymOps) {
+        unsigned symOpNum = operandNo - op->getNumOperands();
+        Operation *symOp = symOps[symOpNum];
+        // Get the verilog name of the operation, add the name if not already
+        // done.
+        if (!names.hasName(symOp)) {
+          StringRef symOpName = getSymOpName(symOp);
```

Comment:
Don't you want to lookup the final name of the entity the symbol refers to?  The ModuleNameManager and getNameRemotely seem the correct thing to do (after ensuring that all names you might lookup have been legalized).  Legalized names are not reflected in the op, getName and symbol name can be wrong.

Reply:
`HWLegalizeModules` should have updated all the names already. So, the `ModuleNameManager `  should have all the final names.